### PR TITLE
Closes #132 removing used SASlength from functions

### LINKED ISSUE #132: xpt_validate attributes not used
I noticed in xpt_validate that ‘SASlength’ is used instead of 'width'. However, this does not have any impact since the ‘SASlength’ attribute is not checked in xpt_validate.

For attribute ‘SAStype’, I ran the function xportr_type(.df, metacore, "test") and did not find the attribute in the dataframe. Furthermore the values check for ‘SAStype’ are based on metadata type ("", "text", "integer", float", "datetime", "date", "time", "partialdate", "partialtime", "partialdatetime", "incompletedatetime", "durationdatetime", "intervaldatetime") and not SAS type ("character" or "numeric")

### LINKED ISSUE #136: Feature Request: Nicer default for 'path' in `xportr_write`
### Feature Idea

Update default for `xportr_write` to be <df_arg>.xpt instead of requiring path to be explicitly passed. Per #107 

### Relevant Input

_No response_

### Relevant Output

_No response_

### Reproducible Example/Pseudo Code

_No response_

### PR DESCRIPTION
**Pull request by sadchla-codes**
### Thank you for your Pull Request!

We have developed a Pull Request template to aid you and our reviewers. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the xportr codebase remains robust and consistent.

### The scope of `{xportr}`

`{xportr}`'s scope is to enable R users to write out submission compliant `xpt` files that can be delivered to a Health Authority or to downstream validation software programs. We see labels, lengths, types, ordering and formats from a dataset specification object (SDTM and ADaM) as being our primary focus. We also see messaging and warnings to users around applying information from the specification file as a primary focus. Please make sure your Pull Request meets this **scope of {xportr}**. If your Pull Request moves beyond this scope, please get in touch with the `{xportr}` team on [slack](https://pharmaverse.slack.com/archives/C030EB2M4GM) or create an issue to discuss.

Please check off each task box as an acknowledgment that you completed the task. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

### Changes Description

_(descriptions of changes)_

### Task List

- [x] The spirit of xportr is met in your Pull Request
- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Summary of changes filled out in the above Changes Description. Can be removed or left blank if changes are minor/self-explanatory.
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Use `styler` package and functions to style files accordingly.
- [x] Updated relevant unit tests or have written new unit tests. See our [Wiki](https://github.com/atorus-research/xportr/wiki/Style-Guide-for-Unit-Tests) for conventions used in this package.
- [x] Creation/updated relevant roxygen headers and examples. See our [Wiki](https://github.com/atorus-research/xportr/wiki/Style-Guide-for-Roxygen-Headers) for conventions used in this package.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Run `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new/updated functions occur on the "Reference" page.
- [x] Update NEWS.md if the changes pertain to a user-facing function (i.e. it has an @export tag) or documentation aimed at users (rather than developers)
- [x] Make sure that the pacakge version in the NEWS.md and DESCRIPTION file is same. Don't worry about updating the version because it will be auto-updated using the `vbump.yaml` CI.
- [x] Address any updates needed for vignettes and/or templates
- [x] Link the issue Development Panel so that it closes after successful merging.
- [x] Fix merge conflicts
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: R/length.R [hunk 1]
```diff
@@ -46,7 +46,7 @@
 #'   the column. This is passed to `haven::write` to note the variable length.
 #'
 #'
-#' @return Data frame with `SASlength` attributes for each variable.
+#' @return Data frame with SAS default length attributes for each variable.
```
Comment (cpiraux) [inline R/length.R:49]:
```suggestion
#' @return Data frame with 'width' attributes that store the SAS length for each variable.
```

File: R/utils-xportr.R [hunk 2]
```diff
@@ -215,7 +215,7 @@ xpt_validate <- function(data) {
   }
 
   # 3.0 VARIABLE TYPES ----
-  types <- tolower(extract_attr(data, attr = "SAStype"))
+  types <- tolower(extract_attr(data, attr = "type"))
```
Comment (cpiraux) [inline R/utils-xportr.R:218]: The `xport_type()` function does not add any attributes, so I suggest to remove the type check.

File: R/write.R [hunk 3]
```diff
@@ -17,11 +17,9 @@
 #' @details
 #'   * Variable and dataset labels are stored in the "label" attribute.
 #'
```
Comment (cpiraux) [inline R/write.R:19]: We can keep the lines for the SAS length and replace 'SASlength' by 'width' :
#' * SAS length are stored in the "width" attribute.

File: tests/testthat/test-utils-xportr.R [hunk 4]
```diff
@@ -85,8 +85,8 @@ test_that("xpt_validate: Get error message when the label contains over 40 chara
 
 test_that("xpt_validate: Get error message when the variable type is invalid", {
   df <- data.frame(A = 1, B = 2)
-  attr(df$A, "SAStype") <- "integer"
-  attr(df$B, "SAStype") <- "list"
+  attr(df$A, "type") <- "integer"
+  attr(df$B, "type") <- "list"
```
Comment (cpiraux) [inline tests/testthat/test-utils-xportr.R:89]: Test to be removed if `xpt_validate()` is updated to remove type check

File: R/write.R [hunk 5]
```diff
@@ -17,11 +17,9 @@
...
-#'   * SAS length are stored in the "SASlength" attribute.
-#'
 #'   * SAS format are stored in the "SASformat" attribute.
```
Comment (cpiraux) [inline R/write.R:20]: The name for the SAS format attribute is 'format.sas'. Maybe we should also update 'SASformat' to  'format.sas', what do you think?

File: R/utils-xportr.R [hunk 6]
```diff
@@ -1,11 +1,11 @@
 #' Extract Attribute From Data
 #'
 #' @param data Dataset to be exported as xpt file
-#' @param attr SAS attributes such as label, format, type, length
+#' @param attr SAS attributes such as label, format, type
 #'
 #' @return Character vector of attributes with column names assigned
 #' @noRd
-extract_attr <- function(data, attr = c("label", "format.sas", "SAStype", "SASlength")) {
+extract_attr <- function(data, attr = c("label", "format.sas", "type")) {
```
Comment (cpiraux) [inline R/utils-xportr.R:8]: There is no "type" attribute added by the function `xport_type()`

**Review by cpiraux — CHANGES_REQUESTED** [review]
Thank you for implementing this. I've included a few comments for your review.

**Review by EeethB — COMMENTED** [review]
No further comments besides Celine's, thanks both for the work on this!

**Comment by elimillera** [comment]
@sadchla-codes Are you able to take care of the rest of those comments?

**Comment by sadchla-codes** [comment]
> @sadchla-codes Are you able to take care of the rest of those comments?

Yes

**Review by cpiraux — APPROVED** [review]
Looks good to me!

**Pull request merged**